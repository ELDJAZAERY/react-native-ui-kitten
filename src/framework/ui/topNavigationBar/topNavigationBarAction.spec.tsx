import React from 'react';
import { TouchableWithoutFeedback } from 'react-native';
import {
  render,
  fireEvent,
} from 'react-native-testing-library';
import {
  styled,
  StyleProvider,
  StyleProviderProps,
} from '@kitten/theme';
import {
  TopNavigationBarAction as TopNavigationBarActionComponent,
  Props as TopNavigationBaActionProps,
} from './topNavigationBarAction.component';
import * as config from './topNavigationBarAction.spec.config';

const TopNavigationBarAction =
  styled<TopNavigationBarActionComponent, TopNavigationBaActionProps>(TopNavigationBarActionComponent);

const Mock = (props?: TopNavigationBaActionProps): React.ReactElement<StyleProviderProps> => (
  <StyleProvider mapping={config.mapping} theme={config.theme} styles={{}}>
    <TopNavigationBarAction {...props} />
  </StyleProvider>
);

const iconSourceUri: string = 'https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png';

describe('@top-navigation-bar-action', () => {

  it('* with icon uri', () => {
    const component = render(<Mock iconSource={{uri: iconSourceUri}}/>);
    expect(component).toMatchSnapshot();
  });

  it('* is last item check', () => {
    const component1 = render(<Mock iconSource={{uri: iconSourceUri}} isLastItem={true}/>);
    const component2 = render(<Mock iconSource={{uri: iconSourceUri}} isLastItem={false}/>);
    expect(component1).toMatchSnapshot();
    expect(component2).toMatchSnapshot();
  });

  it('* on press check', () => {
    const onPress = jest.fn();
    const component = render(<Mock iconSource={{uri: iconSourceUri}} onPress={onPress}/>);

    fireEvent.press(component.getByType(TouchableWithoutFeedback));
    expect(onPress).toHaveBeenCalled();
    expect(component).toMatchSnapshot();
  });

});
