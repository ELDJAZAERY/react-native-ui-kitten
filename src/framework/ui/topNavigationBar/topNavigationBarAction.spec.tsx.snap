// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@top-navigation-bar-action * is last item check 1`] = `
<View
  accessible={true}
  appearance="default"
  dispatch={[Function]}
  iconSource={
    Object {
      "uri": "https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png",
    }
  }
  isLastItem={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "height": 25,
        "marginRight": 0,
        "width": 25,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "height": 25,
      "marginRight": 8,
      "width": 25,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png",
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`@top-navigation-bar-action * is last item check 2`] = `
<View
  accessible={true}
  appearance="default"
  dispatch={[Function]}
  iconSource={
    Object {
      "uri": "https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png",
    }
  }
  isLastItem={false}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "height": 25,
        "marginRight": 8,
        "width": 25,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "height": 25,
      "marginRight": 8,
      "width": 25,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png",
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`@top-navigation-bar-action * on press check 1`] = `
<View
  accessible={true}
  appearance="default"
  dispatch={[Function]}
  iconSource={
    Object {
      "uri": "https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png",
    }
  }
  isLastItem={false}
  onPress={
    [MockFunction] {
      "calls": Array [
        Array [],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "height": 25,
        "marginRight": 8,
        "width": 25,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "height": 25,
      "marginRight": 8,
      "width": 25,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png",
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;

exports[`@top-navigation-bar-action * with icon uri 1`] = `
<View
  accessible={true}
  appearance="default"
  dispatch={[Function]}
  iconSource={
    Object {
      "uri": "https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png",
    }
  }
  isLastItem={false}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Array [
      Object {
        "height": 25,
        "marginRight": 8,
        "width": 25,
      },
      undefined,
    ]
  }
  theme={
    Object {
      "blue-dark": "#2541CC",
      "blue-primary": "#3366FF",
      "gray-dark": "#8992A3",
      "gray-highlight": "#EDF0F5",
      "gray-light": "#DDE1EB",
      "gray-primary": "#A6AEBD",
      "pink-primary": "#FF3D71",
    }
  }
  themedStyle={
    Object {
      "height": 25,
      "marginRight": 8,
      "width": 25,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://pngimage.net/wp-content/uploads/2018/05/back-icon-png-6.png",
      }
    }
    style={
      Object {
        "flex": 1,
      }
    }
  />
</View>
`;
